src/nucleus/tt.ml

```diff
@@ -569,6 +583,68 @@ and print_term' ~penv ?max_level e ppf =
 
 and print_ty ?max_level ~penv (Ty t) ppf = print_term ?max_level ~penv t ppf
 
+(** [print_app e1 e2 ppf] prints the application [e1 e2] using formatter [ppf],
+    possibly as a prefix or infix operator. *)
+and print_app ?max_level ~penv e1 e2 ppf =
+  let e1_prefix =
+    match e1.term with
+    | Bound k ->
+       begin
+         match List.nth penv.forbidden k with
+         | Name.Ident (_, Name.Prefix) as op -> Some (As_ident op)
+         | Name.Ident (_, _) -> None
+       end
+    | Atom (Name.Atom (_, Name.Prefix, _) as op) -> Some (As_atom op)
+    | Constant (Name.Ident (_, Name.Prefix) as op) -> Some (As_ident op)
+    | _ -> None
+  in
+  match e1_prefix with
+  | Some (As_atom op) ->
+     Print.print ppf ?max_level ~at_level:Lvl.prefix "%t@ %t"
+                 (Name.print_atom ~parentheses:false op)
+                 (print_term ~max_level:Lvl.prefix_arg ~penv e2)
+
+  | Some (As_ident op) ->
+     Print.print ppf ?max_level ~at_level:Lvl.prefix "%t@ %t"
+                 (Name.print_ident ~parentheses:false op)
+                 (print_term ~max_level:Lvl.prefix_arg ~penv e2)
+
+  | None ->
+     (* Infix or ordinary application. *)
+     begin
+       let e1_infix =
+         begin
+           match e1.term with
+           | Apply ({term=Constant (Name.Ident (_,
+                                                ((Name.Infix0 | Name.Infix1 | Name.Infix2|
+                                                  Name.Infix3 | Name.Infix4) as fixity)) as op);_},
+                    _, e1) ->
+              Some (As_ident op, fixity, e1)
+           | Apply ({term=Atom (Name.Atom (_,
+                                           ((Name.Infix0 | Name.Infix1 | Name.Infix2|
+                                             Name.Infix3 | Name.Infix4) as fixity), _) as op);_},
+                    _, e1) ->
+              Some (As_atom op, fixity, e1)
+           | _ -> None
```

Comment:
Couldn't it be `Apply (Bound k, _, _)` where the kth bound is infix?
I think there should be a function `val fixity_of_term : term -> (name_printing * Name.fixity) option` 
Then 
```
e1_infix = match e1.term with
  | Apply (h,_,e1) -> begin match fixity_of_term h with
    | Some (op,((Name.Infix0 | ... | Name.Infix4) as fixity)) -> Some (op,fixity,e1)
    | _ -> None
    end
  | _ -> None
```